test/cli-validator/tests/circular-refs-validator.test.js

```diff
@@ -87,4 +87,28 @@ describe('cli tool - test circular reference module', function() {
       )
     ).toEqual(true);
   });
+
+  it('should not fail when a multi-file spec contains circular references across files', async function() {
+    const consoleSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+
+    const program = {};
+    program.args = ['./test/cli-validator/mockFiles/multi-file-spec/main.yaml'];
+    program.default_mode = true;
+    program.ruleset = 'test/spectral/mockFiles/mockConfig/extends-default.yaml';
+
+    const exitCode = await commandLineValidator(program);
+
+    const capturedText = getCapturedText(consoleSpy.mock.calls);
+    consoleSpy.mockRestore();
+
+    expect(exitCode).toEqual(0);
+
+    const allOutput = capturedText.join('');
+
+    expect(
+      allOutput.includes(
+        'Swagger object should not contain circular references.'
```

Comment:
It would be nice to see in the error where circular references have been found.

Reply:
@Legion112 It is in the error, it's just not being checked in the test. Agreed though, that would be a good thing to test here. I'll add that